lib/Sema/SemaBounds.cpp

```diff
@@ -262,12 +330,48 @@ namespace {
                                              SourceLocation());
     }
 
+    // This describes a bounds of empty range. We use this
+    // for example for function pointers.
+    BoundsExpr *CreateBoundsEmpty() {
+      return CreateBoundsNone();
+    }
+
+    // This describes that this is an expression we will never
+    // be able to infer bounds for.
+    BoundsExpr *CreateBoundsUnknown() {
+      return CreateBoundsNone();
+    }
+
+    // If we have an error in our bounds inference that we can't
+    // recover from, bounds(none) is our error value
+    BoundsExpr *CreateBoundsInferenceError() {
+      return CreateBoundsNone();
+    }
+
+    // This describes the bounds of null, which is compatible with every
+    // other bounds annotation.
     BoundsExpr *CreateBoundsAny() {
       return new (Context) NullaryBoundsExpr(BoundsExpr::Kind::Any,
                                              SourceLocation(),
                                              SourceLocation());
     }
 
+    // Currently our inference algorithm has some limitations,
+    // where we cannot express bounds for things that will have bounds
+    //
+    // This is for the case where we want to allow these today,
+    // but we need to re-visit these places and disallow some instances
+    // when we can accurately calculate these bounds.
+    BoundsExpr *CreateBoundsAllowedButUncomputable() {
```

Comment:
I'm changing this to `CreateBoundsAllowedButNotComputed()` - the original intent was to go for something along the lines of "this is not expressible/computable" but I'm happy with the above